Sources/SwiftDocC/Infrastructure/Symbol Graph/GeneratedDocumentationTopics.swift

```diff
@@ -40,25 +40,35 @@ enum GeneratedDocumentationTopics {
         ///   - reference: The parent type reference.
         ///   - originDisplayName: The origin display name as provided by the symbol graph.
         ///   - extendedModuleName: Extended module name.
-        mutating func add(_ childReference: ResolvedTopicReference, to reference: ResolvedTopicReference, originDisplayName: String, extendedModuleName: String) throws {
-            // Detect the path components of the providing the default implementation.
-            let typeComponents = originDisplayName.components(separatedBy: ".")
-            
-            // Verify that the fully qualified name contains at least a type name and default implementation name.
-            guard typeComponents.count >= 2 else { return }
-            
+        mutating func add(_ childReference: ResolvedTopicReference, to reference: ResolvedTopicReference, originDisplayName: String, originParentSymbol: ResolvedTopicReference?, extendedModuleName: String) throws {
+            let fromType: String
+            let typeSimpleName: String
+            if let originParentSymbol = originParentSymbol, !originParentSymbol.pathComponents.isEmpty {
+                // If we have a resolved symbol for the parent of `sourceOrigin`, use that for the names
+                fromType = originParentSymbol.pathComponents.joined(separator: ".")
+                typeSimpleName = originParentSymbol.pathComponents.last!
+            } else {
+                // If we don't have a resolved `sourceOrigin` parent, fall back to parsing its display name
+
+                // Detect the path components of the providing the default implementation.
+                let typeComponents = originDisplayName.components(separatedBy: ".").filter({ !$0.isEmpty })
```

Comment:
Nit: The default behavior of the Swift native `String.split` actually does this filtering for empty components automatically: https://developer.apple.com/documentation/swift/string/split(separator:maxsplits:omittingemptysubsequences:)

```suggestion
                let typeComponents = originDisplayName.split(separator: ".")
```